dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1479,180 +1465,189 @@ synchronized String evaluateInterval() {
         long deltaTime = Math.max(currentTime - lastTimerPop, interval);
         long deltaCompleted = completedWork - previousCompleted;
         double throughput = 1000.0 * deltaCompleted / deltaTime;
+        queueDepth = threadPool.getQueue().size();
+        activeThreads = threadPool.getActiveCount();
+        boolean queueEmpty = (queueDepth <= 0);
+        // Count the number of consecutive times we've seen an empty queue
+        if (!queueEmpty) {
+            consecutiveQueueEmptyCount = 0;
+        } else if (lastAction != LastAction.SHRINK) { // 9/5/2012
+            consecutiveQueueEmptyCount++;
+        }
```

Comment:
The empty queue information isn't used in the trace statement so does it need to be outside of the if statement?

Reply:
Well `consecutiveQueueEmptyCount` is reported in the trace output ... 

`        sb.append("\nHeuristics:");
        sb.append(String.format(" queueDepth = %8d", Integer.valueOf(queueDepth)));
        sb.append(String.format(" consecutiveQueueEmptyCount = %2d", Integer.valueOf(consecutiveQueueEmptyCount)));
`
but it probably does not does not add much diagnostic value, in the case `coreThreads == maxThreads` as long as `queueDepth` is reported. 

So I've moved the lines from `boolean queueEmpty ...` through the end of the highlighted section inside the if statement, with the `deepQueue` calculation.

Reply:
I've now made changes to `toIntervalData` to exclude irrelevant items from the output when `coreThreads == maxThreads` which should make the above `queueEmpty` values a non-issue, and has the added benefit of reducing pathlength when trace is active.